conan/tools/ros/rosenv.py

```diff
@@ -0,0 +1,49 @@
+import os
+from conan.api.output import Color
+from conan.tools.env import VirtualBuildEnv, Environment
+from conan.tools.files import save
+
+
+class ROSEnv:
+    """
+    Generator to serve as integration for Robot Operating System 2 development workspaces.
+    It generates a conanrosenv.sh file that when sources sets variables so the Conan
+    dependencies are found by CMake and the run environment is also set.
+
+    IMPORTANT: This generator should be used together with CMakeDeps and CMakeToolchain generators.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._variables = {}
+        self.variables = {}
+        self._virtualbuildenv = VirtualBuildEnv(self._conanfile, auto_generate=True)
+        self._virtualbuildenv.basename = "conanrosenv"
+        self._rosenv_wrapper = "conanrosenv.sh"
+
+    def generate(self):
+        output_folder = self._conanfile.generators_folder
+        self._variables["CMAKE_TOOLCHAIN_FILE"] = os.path.join(output_folder, "conan_toolchain.cmake")
+        build_type = self._conanfile.settings.get_safe("build_type")
+        if build_type:
+            self._variables["CMAKE_BUILD_TYPE"] = build_type
+        self.variables.update(self._variables)
+
+        # Add ROS required variables to VirtualBuildEnv
+        rosbuildenv = Environment()
+        for k, v in self.variables.items():
+            rosbuildenv.define(k, v)
+        self._virtualbuildenv._buildenv = rosbuildenv
+        self._virtualbuildenv.generate()
+
+        # Generate conanrosenv.sh script wrapper that calls conanbuild.sh and conanrun.sh
+        # TODO: Windows .bat/.ps1 files still not supported for the wrapper
+        conanbuild_path = os.path.join(self._conanfile.generators_folder, "conanbuild.sh")
+        conanrun_path = os.path.join(self._conanfile.generators_folder, "conanrun.sh")
+        rosenv_wrapper_content = [f". \"{conanbuild_path}\"", f". \"{conanrun_path}\""]
+        conanrosenv_path = os.path.join(self._conanfile.generators_folder, self._rosenv_wrapper)
+        save(self._conanfile, conanrosenv_path, "\n".join(rosenv_wrapper_content))
```

Comment:
Use the ``create_env_script()`` functionality instead, so it registers the ``rosenv`` into the build scripts (it will be called by ``conanbuild.sh``)